### ISSUE TITLE
Implement cyclic learning rate schedule

### ISSUE THREAD

**Issue by epwalsh** [epwalsh/pytorch-crf#75]
- [x] Implement basic cyclic learning rate schedules [https://arxiv.org/abs/1506.01186](https://arxiv.org/abs/1506.01186)
- [x] Implement multiplier option
- [x] Write tests for SGD optimizer
